src/prefect/engine/task_runner.py

```diff
@@ -713,9 +709,9 @@ def run_mapped_task(
                                         preview=repr(upstream_state.result)[:10],
                                     )
                                 )
-                            upstream_result = Result(
+                            upstream_result = Result(  # todo: here
```

Comment:
Not super confident what to be doing here (if it's more like the task inputs/cached inputs checks, or more like the loops) so would appreciate some more discussion on this.